static/app/views/performance/transactionDetails/content.tsx

```diff
@@ -59,58 +61,36 @@ type Props = Pick<RouteComponentProps<{eventSlug: string}, {}>, 'params' | 'loca
     projects: Project[];
   };
 
-type State = {
-  event: Event | undefined;
-  isSidebarVisible: boolean;
-} & DeprecatedAsyncComponent['state'];
+function EventDetailsContent(props: Props) {
+  const [isSidebarVisible, setIsSidebarVisible] = useState<boolean>(true);
+  const projectId = props.eventSlug.split(':')[0];
+  const {organization, eventSlug, location} = props;
 
-class EventDetailsContent extends DeprecatedAsyncComponent<Props, State> {
-  state: State = {
-    // AsyncComponent state
-    loading: true,
-    reloading: false,
-    error: false,
-    errors: {},
-    event: undefined,
+  const {
+    data: event,
+    isLoading,
+    error,
+  } = useApiQuery<Event>(
+    [`/organizations/${organization.slug}/events/${eventSlug}/`],
+    {staleTime: 2 * 60 * 1000} // 2 minutes in milliseonds
+  );
 
-    // local state
-    isSidebarVisible: true,
-  };
-
-  toggleSidebar = () => {
-    this.setState({isSidebarVisible: !this.state.isSidebarVisible});
-  };
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization, params} = this.props;
-    const {eventSlug} = params;
-
-    const url = `/organizations/${organization.slug}/events/${eventSlug}/`;
-
-    return [['event', url]];
-  }
-
-  onLoadAllEndpointsSuccess() {
-    const {location, projects} = this.props;
-    const {event} = this.state;
-    this.props.setEventNames(
-      'performance.event_details',
-      'Performance: Opened Event Details'
-    );
-    this.props.setRouteAnalyticsParams({
-      event_type: event?.type,
-      project_platforms: getSelectedProjectPlatforms(location, projects),
-      ...getAnalyticsDataForEvent(event),
-    });
-  }
-
-  get projectId() {
-    return this.props.eventSlug.split(':')[0];
-  }
+  useEffect(() => {
+    if (event) {
+      const {projects} = props;
+      props.setEventNames(
+        'performance.event_details',
+        'Performance: Opened Event Details'
+      );
+      props.setRouteAnalyticsParams({
+        event_type: event?.type,
+        project_platforms: getSelectedProjectPlatforms(location, projects),
+        ...getAnalyticsDataForEvent(event),
+      });
+    }
+  }, [event, props, location]);
 
-  generateTagUrl = (tag: EventTag) => {
-    const {location, organization} = this.props;
-    const {event} = this.state;
+  const generateTagUrl = (tag: EventTag) => {
```

Comment:
We might want to wrap this function (e.g. `renderBody`) and the other ones that are generated with a `useCallback` so it isn't being redefined every time this component re-renders